presto-iceberg/src/main/java/com/facebook/presto/iceberg/HiveTableOperations.java

```diff
@@ -88,6 +91,9 @@ public class HiveTableOperations
     public static final String PREVIOUS_METADATA_LOCATION = "previous_metadata_location";
     private static final String METADATA_FOLDER_NAME = "metadata";
 
+    @VisibleForTesting
+    protected static Supplier<RetryConfig> defaultRetrySupplier = RetryConfig::new;
```

Comment:
Should we make `defaultRetrySupplier` thread local to ensure thread safety? As the tests may run concurrently, and we might add other tests which set `RetryConfig` as well.

Reply:
Even if tests run concurrently, setting this as a thread-local probably won't solve the issue because we don't actually know which threads is going to end up creating the `HiveTableOperations`, since the query runner executes across a large number of threads.

Reply:
I'm not so sure but as I understand `HiveTableOperations ` is not a global attribute in query runner, and its lifecycle would not exceed a single transaction. So that the creating and using of `HiveTableOperations` could be thread safe. Its not important anymore if we plan to add the configs to IcebergConfig as @tdcmeehan and your issue #21967 has mentioned.

Reply:
Yes, that's exactly what I meant.